WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingCategoriesAdapter.kt

```diff
@@ -1,121 +1,77 @@
-import org.wordpress.android.R
-import org.wordpress.android.models.CategoryNode
-import org.wordpress.android.ui.posts.PrepublishingCategoriesAdapter.CategoriesViewHolder
+import org.wordpress.android.ui.posts.PrepublishingCategoriesViewHolder.PrepublishingCategoriesListItemViewHolder
+import org.wordpress.android.ui.posts.PrepublishingCategoriesViewModel.PrepublishingCategoriesListItemUiState
+import org.wordpress.android.ui.utils.UiHelpers
 
-class PrepublishingCategoriesAdapter(
-    private val context: Context,
-    private val onCheckChangeListener: ((Long, Boolean) -> Unit)
-) : RecyclerView.Adapter<CategoriesViewHolder>() {
-    var categoryNodeList: List<CategoryNode> = arrayListOf()
-        set(value) {
-            if (!isSameCategoryList(value)) {
-                field = value
-                notifyDataSetChanged()
-            }
-        }
-
-    private var selectedCategoryIds: HashSet<Long> = hashSetOf()
-
-    fun set(categoryLevels: List<CategoryNode>, categoriesSelected: HashSet<Long>) {
-        selectedCategoryIds = categoriesSelected
-        categoryNodeList = categoryLevels
-    }
-
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CategoriesViewHolder {
-        val itemView = LayoutInflater.from(context)
-                .inflate(R.layout.prepublishing_categories_row, parent, false)
-        return CategoriesViewHolder(onCheckChangeListener, itemView)
-    }
-
-    override fun onBindViewHolder(viewHolder: CategoriesViewHolder, position: Int) {
-        viewHolder.bind(categoryNodeList[position], selectedCategoryIds)
-    }
+class PrepublishingCategoriesAdapter(private val uiHelpers: UiHelpers) :
+        RecyclerView.Adapter<PrepublishingCategoriesViewHolder>() {
+    private val items = mutableListOf<PrepublishingCategoriesListItemUiState>()
 
     init {
         setHasStableIds(true)
     }
 
-    override fun getItemId(position: Int): Long {
-        return categoryNodeList[position].categoryId
-    }
-
-    override fun getItemCount(): Int {
-        return categoryNodeList.size
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ): PrepublishingCategoriesViewHolder {
+        return PrepublishingCategoriesListItemViewHolder(parent, uiHelpers)
     }
 
-    private fun isSameCategoryList(categoryList: List<CategoryNode>): Boolean {
-        if (categoryList.size != categoryNodeList.size) {
-            return false
-        }
+    override fun getItemCount(): Int = items.size
 
-        categoryNodeList.forEach {
-            if (!containsNode(it)) {
-                return false
-            }
-        }
-        return true
+    override fun getItemId(position: Int): Long {
+        return position.toLong()
     }
 
-    private fun isSameSelectedList(selectedCategoryIdList: HashSet<Long>): Boolean {
-        return selectedCategoryIdList == selectedCategoryIds
+    override fun getItemViewType(position: Int): Int {
+        return position
```

Comment:
AFAIU this means that each item has it's own viewType - is this really what we want?

Reply:
> AFAIU this means that each item has it's own viewType - is this really what we want?

Nope, this is not what I wanted. Just a left over toying with another idea. Gone in a983929